fix_proto_conf_desc.py

```diff
@@ -0,0 +1,170 @@
+import ida_segment
+import idautils
+import idc
+import ida_bytes
+import ida_name
+from ida_nalt import REF_OFF32, REFINFO_SIGNEDOP, REFINFO_SELFREF
+from idc import FF_DATA, FF_DWORD
+
+class CStruct: 
+    SELFREF = REFINFO_SIGNEDOP | REFINFO_SELFREF
+
+    def __init__(self):
+        self.tif = 0
+
+    def _append_member(self, name: str, ida_type: int) -> ida_typeinf.udm_t:
+        """
+        Append a udm_t into `self.tif` and return it so user is able to add his
+        own modifications.
+        """
+        udm = ida_typeinf.udm_t()
+        udm.name = name
+        udm.offset = self.tif.get_unpadded_size() * 8
+        udm.type = ida_typeinf.tinfo_t(ida_type)
+        udm.size = udm.type.get_size() * 8
+        return udm
+
+    def append_int16_member(self, name: str):
+        udm = self._append_member(name, ida_typeinf.BT_INT16)
+        self.tif.add_udm(udm)
+
+    def append_int32_member(self, name: str):
+        udm = self._append_member(name, ida_typeinf.BT_INT32)
+        self.tif.add_udm(udm)
+
+    def append_reloffset_member(self, name: str) -> None:
+        udm = self._append_member(name, ida_typeinf.BT_INT32)
+        udm.repr.set_vtype(ida_typeinf.FRB_OFFSET)
+        udm.repr.ri.init(ida_nalt.REF_OFF32 | self.SELFREF)
+        self.tif.add_udm(udm)
+
+    def get_name(self):
+        return self.tif.get_type_name()
+
+    def get_size(self):
+        return self.tif.get_unpadded_size()
+
+class PCDStruct(CStruct): 
+    PCD_BASE_IDA_STRUCT_TYPE_NAME = "ProtConfDescriptor"
+
+    def _create_pcd(self, pcd_address: int):
+        self.tif = ida_typeinf.tinfo_t()
+        udt = ida_typeinf.udt_type_data_t()
+        self.tif.create_udt(udt)
+        self.tif.set_named_type(None, f"{self.PCD_BASE_IDA_STRUCT_TYPE_NAME}_{pcd_address:x}")
+        self.append_reloffset_member("ProtocolDescriptor")
+        self.append_reloffset_member("NominalTypeDescriptor")
+        #If PWT field creates a reloffset to itself (PWT dword = 0) it means it doesn't have PWT.
+        self.append_reloffset_member("PWT")
+        self.append_int32_member("Flags")
+
+    def __init__(self, pcd_address: int):  
+        self._create_pcd(pcd_address)
```

Comment:
Trigger the super's ctor. Then u won't need to re-init `self.tif`
```suggestion
        super().__init__()
        self._create_pcd(pcd_address)
```